File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/setshapefilldata.rb
module Asposediagramjava
  module SetShapeFillData
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        shapes = diagram.getPages().getPage(0).getShapes()
        
        i = 0
        while i < shapes.getCount()
            shape = shapes.get(i)
            if shape.getNameU() == "rectangle" && shape.getID() == 1
                shape.getFill().getFillBkgnd().setValue(diagram.getPages().getPage(0).getShapes().getShape(0).getFill().getFillBkgnd().getValue())
                shape.getFill().getFillForegnd().setValue("#ebf8df")
            end
            i +=1
        end
        
        # Save diagram
        diagram.save(data_dir + "SetShapeFillData.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Set visio shape's fill data."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/SetShapeFillData.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class SetShapeFillData{

    public static function run($dataDir=null){

    # Create instance of Diagram
    $diagram = new Diagram($dataDir."Drawing.vsd");

    $shapes=$diagram->getPages()->getPage(0)->getShapes();

    $i = 0;
    while ($i<sizeof($shapes->getCount())) {
        $shape = $shapes->get($i);
        if ($shape->getNameU() == "rectangle" && $shape->getID() == 1) {
        $shape->getFill()->getFillBkgnd()->setValue($diagram->getPages()->getPage(0)->getShapes()->getShape(0)->getFill()->getFillBkgnd()->getValue());
        $shape->getFill()->getFillForegnd()->setValue("#ebf8df");
        }
        $i += 1;
    }

    # Save diagram
    $saveFileFormat=new SaveFileFormat();
    $diagram->save($dataDir."SetShapeFillData.vdx",$saveFileFormat->VDX);
    print "Set visio shape's fill data.".PHP_EOL;
    }
}